ws-server/src/tests.rs

```diff
@@ -202,3 +232,24 @@ async fn invalid_request_should_not_close_connection() {
 	let response = client.send_request_text(request).await.unwrap();
 	assert_eq!(response, ok_response(JsonValue::String("hello".to_owned()), Id::Num(33)));
 }
+
+#[tokio::test]
+async fn valid_request_that_fails_to_execute_should_not_close_connection() {
+	let addr = server().await;
+	let mut client = WebSocketTestClient::new(addr).await.unwrap();
+
+	// Good request, executes fine
+	let request = r#"{"jsonrpc":"2.0","method":"say_hello","id":33}"#;
+	let response = client.send_request_text(request).await.unwrap();
+	assert_eq!(response, ok_response(JsonValue::String("hello".to_owned()), Id::Num(33)));
+
+	// Good request, but causes error.
```

Comment:
```suggestion
	// Good request with unexpected params
```

Reply:
No, the point here is to show that a **valid** request object that still fails to execute correctly comes back with the expected (app-level) error. And that this doesn't take down the server ofc.

Reply:
ah, I see missed that it was an error,